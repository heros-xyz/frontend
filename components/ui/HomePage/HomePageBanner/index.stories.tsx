import type { Meta, StoryObj } from "@storybook/react";
import HomePageBanner from "./index";

const meta: Meta<typeof HomePageBanner> = {
  title: "HomePage/HomePageBanner",
  component: HomePageBanner,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof HomePageBanner>;

export const Banner: Story = {
  args: {
    title: "For athletes, by fans",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
  },
};
