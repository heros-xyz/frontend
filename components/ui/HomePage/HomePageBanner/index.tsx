import { Box, Button, Flex, Heading, Image, Text } from "@chakra-ui/react";
import Link from "next/link";
import React from "react";

interface HomePageBannerProps {
  content: string;
  title: string;
}

const HomePageBanner: React.FC<HomePageBannerProps> = ({ content, title }) => {
  return (
    <Box>
      <Box>
        <Image
          src="/images/HomePageBanner.png"
          alt="Home Page Banner"
          width="100%"
          height="100%"
        />
      </Box>
      <Box bg="accent.4" borderRadius="0px 0px 18px 18px" px="5" py="7">
        <Heading
          color="accent.2"
          fontSize="2xl"
          fontWeight="extrabold"
          lineHeight="140%"
          textTransform="capitalize"
        >
          {title}
        </Heading>
        <Text
          mt="2"
          mb="7"
          fontSize="sm"
          fontWeight="normal"
          color="primary"
          lineHeight="140%"
        >
          {content}
        </Text>
        <Flex justify="space-around">
          <Link href="/fan/sign-up">
            <Button variant="primary" height="48px" mr="2.5">
              Join as Fan
            </Button>
          </Link>
          <Link href="/athlete/sign-up">
            <Button variant="primary" height="48px">
              Join as athlete
            </Button>
          </Link>
        </Flex>
      </Box>
    </Box>
  );
};

export default HomePageBanner;
