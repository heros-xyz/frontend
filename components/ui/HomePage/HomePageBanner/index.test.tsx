import { render } from "@testing-library/react";
import React from "react";
import HomePageBanner from "./index";

it("renders homepage banner unchanged", () => {
  const { container } = render(
    <HomePageBanner
      title="For athletes, by fans"
      content="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
    />
  );
  expect(container).toMatchSnapshot();
});
