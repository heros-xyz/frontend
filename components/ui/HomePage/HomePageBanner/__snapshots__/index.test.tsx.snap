// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage banner unchanged 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-0"
    >
      <img
        alt="Home Page Banner"
        class="chakra-image css-uwwqev"
        src="/images/HomePageBanner.png"
      />
    </div>
    <div
      class="css-qr35ly"
    >
      <h2
        class="chakra-heading css-1hdztj9"
      >
        For athletes, by fans
      </h2>
      <p
        class="chakra-text css-26z8ad"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
      <div
        class="css-1bnxa12"
      >
        <a
          href="/fan/sign-up"
        >
          <button
            class="chakra-button css-nas344"
            type="button"
          >
            Join as Fan
          </button>
        </a>
        <a
          href="/athlete/sign-up"
        >
          <button
            class="chakra-button css-lrwttc"
            type="button"
          >
            Join as athlete
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
